### PR TITLE
WP 5.5 Compat: Align Jetpack and Core's plugin autoupdates

### LINKED ISSUE #16600: WP 5.5 Autoupdates: Make JP/Core autoupdates compatible
WordPress 5.5 is releasing UI and underlying support for per-plugin and per-theme autoupdates.

Jetpack, via Calypso and Calypsoify, offers this feature too. We can merge the features to utilize the same option and avoid duplication.

- [x] Remove the Calypsoify auto-update UI.
- [x] Migrate Jetpack option to use Core's option. (`Jetpack_Options::get_option( 'autoupdate_plugins', array() );` -> `get_site_option( 'auto_update_plugins', array() );`
- [x] Sync Core's option.
- [x] Map WP.com to save Core's option where the existing Jetpack option resides (to avoid a lot of code churn and option updating on the WP.com side.
- [x] Update the REST API for the new option.
- [ ] Verify Calypso works with the REST API update.
- [ ] Open Question: Disable Core's processing of the auto-update to rely on Jetpack?

WordPress 5.5 compat issue: https://github.com/Automattic/jetpack/issues/15388

### PR DESCRIPTION
**Pull request by kraftbj**
Fixes #16600
See #15388

#### Changes proposed in this Pull Request:
* Convert the existing `Jetpack_Option` autoupdate plugin list to use new Core option. This means existing autoupdating plugins will be included in the Core update.
* Sync the network option.
* Enable listeners for network options off of multisite. They aren't exclusive to multisite and are, in effect, the same as regular options when used on a single site.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
n/a

#### Testing instructions:
* Existing Jetpack site before this patch running on WP 5.4.
* Visit wp-admin/plugins.php?calypsoify=1
* Enable plugin autoupdates for something.
* Via wp.com sandbox, get the `jetpack_autoupdate_plugins` option. Verify it matches.
* Upgrade to WP 5.5 (via the WP beta tester).
* Upgrade to this patch.
* Visit the wp-admin/plugins.php?calypsoify=0 and confirm the Core UI shows the same plugins enabled.
* Visit wp-admin/plugins.php?calypsoify=1 and ensure you're only seeing on option to auto-update (the Core version).
* Make changes.
* Via wp.com sandbox, get the `jetpack_autoupdate_plugins` option. Verify it matches.

#### Proposed changelog entry for your changes:
* TBD

### CODE REVIEW

File: tests/php/sync/test_class.jetpack-sync-plugins.php [hunk 1]
```diff
@@ -61,24 +61,24 @@ public function test_installing_and_removing_plugin_is_synced() {
 
 	public function test_autoupdate_enabled_and_disabled_is_synced() {
 		// enable autoupdates
-		$autoupdate_plugins = Jetpack_Options::get_option( 'autoupdate_plugins', array() );
+		$autoupdate_plugins = (array) get_site_option( 'auto_update_plugins', array() );
 		$autoupdate_plugins = array_unique( array_merge( $autoupdate_plugins, array( 'hello' ) ) );
-		Jetpack_Options::update_option( 'autoupdate_plugins', $autoupdate_plugins );
+		update_site_option( 'auto_update_plugins', $autoupdate_plugins );
 		$this->sender->do_sync();
```
Comment (mdbitz) [inline tests/php/sync/test_class.jetpack-sync-plugins.php:67]: As this option isn't part of the always send "critical" options we need to make an explicit call to unlock callables sync. 
`do_action( 'jetpack_sync_unlock_sync_callable' );`

Comment (mdbitz) [reply tests/php/sync/test_class.jetpack-sync-plugins.php:67]: if we do want to make this a critical option then we would need to make a few changes to class-callables.php in sync/modules

1) add "auto_update_plugins" to ALWAYS_SEND_UPDATES_TO_THESE_OPTIONS const
2) add "'auto_update_plugins' => 'autoupdate_plugins'," to OPTION_NAMES_TO_CALLABLE_NAMES
3) within init_listeners, we would need to update the foreach to include hooks to 'update_site_option_{$option}' and 'delete_site_option_{$option}'

**Comment by kraftbj** [comment]
The WP.com side for processing the option name shipped in r211863-wpcom

File: tests/php/sync/test_class.jetpack-sync-plugins.php (hunk 1 above)
Comment (kraftbj) [reply tests/php/sync/test_class.jetpack-sync-plugins.php:67]: Ended up going a slightly different route just using the network option directly, but had to disable the multisite checks. `site_options` aren't exclusive to mulitisite installs—just not really used much for single sites.

**Comment by jeherve** [comment]
Looks like the Legacy Sync test keeps failing.

**Comment by kraftbj** [comment]
@jeherve Sorry about that! All green now.

**Comment by jeherve** [comment]
Cherry-picked to `branch-8.8` in 9550f0e02971c12702fd185d28d294ea23412fd3

**Comment by jeherve** [comment]
Updated the Sync package in the release branch in bc69c8088c7f53dc3e5b94c588a01cc2b18a7e45

**Pull request merged**